src/object/srv_obj.c

```diff
@@ -1375,11 +1376,12 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 	rma = (orw->orw_bulks.ca_arrays != NULL ||
 	       orw->orw_bulks.ca_count != 0);
 	cond_flags = orw->orw_api_flags;
-
+	if (daos_oclass_is_ec(&ioc->ioc_oca))
+		tgt_off = obj_ec_shard_off_by_layout_ver(ioc->ioc_layout_ver, orw->orw_dkey_hash,
+							 &ioc->ioc_oca, orw->orw_oid.id_shard);
```

Comment:
tgt_off seems is same as orw_tgt_idx? can just use orw_tgt_idx, or assert(tgt_off == orw_tgt_idx)?

Reply:
I will check this in the following patch.